homeassistant/components/amcrest/__init__.py

```diff
@@ -184,6 +200,38 @@ def _wrap_test_online(self, now):
             pass
 
 
+def _monitor_events(hass, name, api, event_codes):
+    event_codes = ",".join(event_codes)
+    start_stop = re.compile(r"Code=([^;]+);action=(Start|Stop)", flags=re.S)
```

Comment:
Can this be parsed without a regex?

Reply:
This parsing should be moved into the pypi package.

Reply:
There is a _LOT_ that should be moved into the pypi package. In fact, IMHO, for purposes of Home Assistant, it should be pretty much rewritten. But that's not likely to happen anytime soon. Also, Amcrest has changed the API of their cameras and has _not_ documented the changes. They won't even admit that the API has changed, let alone provide any documentation. Newer cameras and newer firmware releases of older cameras are becoming incompatible with the pypi package and Home Assistant. I'm not sure it's worth the effort to make any changes at this point to the pypi package. At least not until Amcrest begins to cooperate again. Is this the only issue with this PR?